### ISSUE TITLE
Inconsistent state-tracking for harbor_replication and harbor_registry resources

### ISSUE THREAD

**Issue by lindhe** [goharbor/terraform-provider-harbor#134]
**Describe the bug**
I have fiddled around a bit with replication and there seems to be some state-tracking issues with the `harbor_replication` and `harbor_registry` resources.

* At least in one scenario, Terraform is unable to change credentials (i.e. Access ID and Access Secret) for a registry endpoints.
* Changing from disabled to enabled in the GUI for a replication is not detected by Terraform.

**To Reproduce**
1. Create a registry endpoint using Terraform, setting `access_id = "foo"`.
1. Go into the web GUI and edit the registry and change Access ID to `bar`.
1. `tf apply`
    1. Verify that the diff is correct (i.e. detecting the change and planning to change it from `bar` to `foo` again since the Terraform configuration says so).
    1. type `yes` to apply plan
1. Run `tf plan` or go to the GUI and see that the Access ID is in fact unchanged (still `bar`).

The replication enable/disable bug can be reproduced similarly.

**Expected behavior**
Terraform should detect the changed state of Harbor and properly apply the desired state again according to the Terraform configuration.

**System Information**
 - Version 3.1.8

**Comment by leviathanbadger** [comment]
It appears that the update model for registries is different from the create model. Create model looks like this:

```typescript
{
  name: string,
  // ...
  credential: {
    access_key: string,
    access_secret: string,
    type: string
  }
}
```

Update model looks like this:
```typescript
{
  name: string,
  // ...
  access_key: string,
  access_secret: string,
  credential_type: string
}
```

I haven't looked at the replication rules, but I'd guess it's a similar issue.

**Comment by naitmare01** [comment]
I have written a PR which _should_ fix this issue. Let me know what you think.

**Comment by flbla** [comment]
should be fixed with v3.8.3
